### ISSUE TITLE
Using naive object to get timezone out of a timestamp should not raise an error

### ISSUE THREAD

**Issue by LefterisJP** [pjknkda/flake8-datetimez#11]
I am not sure if this is the right way to do it. but if you want to get a datetime from a timestamp and be aware of the timestamp's timezone (especially whether it's DST or not) you need to do the following (if there is a better/smarter way please let me know because this feels weird). This is in my locale time CET/CEST.

```python
>>> formatstr = '%d/%m/%Y %H:%M:%S %Z'
>>> ts = 1675261083
>>> datetime.datetime.fromtimestamp(ts, tz=datetime.datetime.fromtimestamp(ts).astimezone().tzinfo).strftime(formatstr)
'01/02/2023 15:18:03 CET'
>>> ts = 1686065064
>>> datetime.datetime.fromtimestamp(ts, tz=datetime.datetime.fromtimestamp(ts).astimezone().tzinfo).strftime(formatstr)
'06/06/2023 17:24:24 CEST
```

This works fine and is correct but flake8-datetimez complains ` DTZ006 The use of `datetime.datetime.fromtimestamp()` without `tz` argument is not allowed`.

I think this is a false positive in this case and it should not complain. I am going to add noqa ignore but perhaps there is a way to code this into the logic of the tool?

**Comment by kbehlers** [comment]
@LefterisJP I believe the warning is legitimate. The call to `.astimezone()` [uses your local system timezone](https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone):
> If called without arguments (or with tz=None) the system local timezone is assumed for the target timezone.

As the warning suggests, it would be best to pass the `tz` argument to `fromtimestamp` so that the output doesn't change based on the local system timezone of the execution environment.

If you are on Python 3.9+, you can use `zoneinfo.Zoneinfo` to do that, otherwise you will need to use a third-party package like [dateutil](https://dateutil.readthedocs.io/en/stable/tz.html).

For your case, that would look something like:
```
>>> from datetime import datetime
>>> import zoneinfo
>>> cet_tz = zoneinfo.ZoneInfo("CET")
>>> datetime.fromtimestamp(ts, tz=cet_tz).strftime(formatstr)
'06/06/2023 17:24:24 CEST'
```

Longer Example:
```
>>> import os
>>> import time
>>> from datetime import datetime
>>> os.environ["TZ"] = "CET"  # python respects the TZ env var
>>> time.tzset()  # https://docs.python.org/3/library/time.html#time.tzset
>>> formatstr = '%d/%m/%Y %H:%M:%S %Z'
>>> ts = 1675261083
>>> datetime.fromtimestamp(ts).astimezone().strftime(formatstr)  # simplified version of original
'01/02/2023 15:18:03 CET'
>>> ts = 1686065064
>>> datetime.fromtimestamp(ts).astimezone().strftime(formatstr)  # simplified version of original
'06/06/2023 17:24:24 CEST'
>>> os.environ["TZ"] = "UTC"
>>> time.tzset()  # https://docs.python.org/3/library/time.html#time.tzset
>>> datetime.fromtimestamp(ts).astimezone().strftime(formatstr)
'06/06/2023 15:24:24 UTC'
>>> import zoneinfo
>>> cet_tz = zoneinfo.ZoneInfo("CET")
>>> datetime.fromtimestamp(ts, tz=cet_tz).strftime(formatstr)
'06/06/2023 17:24:24 CEST'  # Despite TZ being UTC, passing tz=cet_tz forces CET time
```
